### DISCUSSION TITLE
Server-Action & preventDefault()

### DISCUSSION THREAD

**Discussion by 2njeong** [vercel/next.js#66401] [Help] [answered]
### Summary

Why Server-Action does not need e.preventDefault() with form tag unlike Client-Side?

I know that it's a basic operation of Form tag to be submitted & reload the page. 
So, on client-side we use e.preventDefault() to prevent those from execution.
But why Server-action does not need e.preventDefault()?
Is the basic operation of Form Tag not operated on Server-Action?
I think it's irrelevent though..

**Comment by icyJoseph** [comment, marked as answer]
For forms, you call, `event.preventDefault` on the `onSubmit` handler, which is a way to stop the default form submission algorithm, that'd pick up action and method props, and use them to submit the form.

Normally, in HTML, the form action has to be a URL, but react-dom has enhanced the form component, by allowing you to pass a function.

If we see this part of the React code:

- https://github.com/facebook/react/blob/63d673c67656390d776bfa082c6ab49f0c636582/packages/react-dom-bindings/src/events/plugins/FormActionEventPlugin.js#L160

We see that, when the form submit event happens, if there's an action function, then react-dom calls preventDefault, for you.

**Reply by 2njeong** [reply]
thanks for great explanation!!
